packages/template-retail-react-app/app/hooks/use-product-pricing.js

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2023, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import {useMemo} from 'react'
+import {getDisplayPrice} from '../utils/product-utils'
+
+/**
+ * @typedef {object} ProductPricing
+ * @property {string} currency
+ * @property {string} basePrice
+ * @property {string} discountPrice
+ */
+
+/**
+ * Gets the product pricing, allowing for extensibility using custom hooks
+ * @param {object} product
+ * @param {number} quantity
+ * @return {ProductPricing}
+ */
+function useProductPricing(product, quantity = 1) {
+    let qty = Number(quantity)
+    if (qty < 1 || isNaN(qty)) {
+        qty = 1
+    }
+    return useMemo(() => {
+        if (product) {
+            const {basePrice, discountPrice} = getDisplayPrice(product)
+            return {
+                currency: product.currency,
+                basePrice: basePrice !== null ? basePrice * qty : null,
+                discountPrice: discountPrice !== null ? discountPrice * qty : null
+            }
+        } else {
+            return {
+                currency: null,
+                basePrice: null,
+                discountPrice: null
+            }
```

Comment:
I'm guessing that you are returning a value under all circumstances so you can do stuff like this `{currency, basePrice, discountPrice} = useProductPricing(product, quantity)` and not run into issues of destructuring a null value?

My question is, does it makes sense that if you don't have a Product object you do have a ProductPricing object. I would have to say no. Even tho the values within the ProductPricing object are null, you still have a pricing object that isn't really associated with any product. 

I'd recommend returning null if you the product being passed in is also null. You can overcome the previous destructuring issue by simply doing `{currency, basePrice, discountPrice} = useProductPricing(product, quantity) || {}`

Reply:
all set